phantom_connector.py

```diff
@@ -534,60 +559,82 @@ def _find_artifacts(self, param):
+                                for y in [x.strip() for x in container_ids.replace(",", " ").split()]
+                            ]
                         ]
-                    ] if a
-                ])
+                        if a
+                    ]
+                )
             )
             action_result.update_param({"container_ids": str(sorted(container_ids)).strip("[]")})
 
         if limit_search and not container_ids:
-            action_result.update_summary({'artifacts_found': 0, 'server': self._base_uri})
+            action_result.update_summary({"artifacts_found": 0, "server": self._base_uri})
             return action_result.set_status(phantom.APP_SUCCESS)
 
         cef_key = param.get("cef_key")
 
-        exact_match = param.get('exact_match', False)
+        exact_match = param.get("exact_match", False)
 
         if exact_match and not cef_key:
             values = '"{}"'.format(values)
 
-        url_enc_values = quote(values, safe='')
+        url_enc_values = quote(values, safe="")
 
         if cef_key and exact_match:
-            endpoint = '/rest/artifact?_filter_cef__{}={}&page_size=0&pretty'.format(quote(cef_key, safe=''), repr(url_enc_values))
+            endpoint = f"/rest/artifact?_filter_cef__{quote(cef_key, safe='')}={repr(url_enc_values)}&pretty"
         elif cef_key:
-            endpoint = '/rest/artifact?_filter_cef__{}__{}={}&page_size=0&pretty'.format(quote(cef_key, safe=''),
-                                                                                        "icontains", repr(url_enc_values))
+            endpoint = f"/rest/artifact?_filter_cef__{quote(cef_key, safe='')}__icontains={repr(url_enc_values)}&pretty"
         else:
-            endpoint = '/rest/artifact?_filter_cef__{}={}&page_size=0&pretty'.format("icontains", repr(url_enc_values))
+            endpoint = f"/rest/artifact?_filter_cef__icontains={repr(url_enc_values)}&pretty"
 
         if limit_search:
-            endpoint += '&_filter_container__in={}'.format(container_ids)
+            endpoint += "&_filter_container__in={}".format(container_ids)
 
-        ret_val, response, resp_data = self._make_rest_call(endpoint, action_result)
+        records = list()
+        page = 0
+        page_size = 10
+        while True:
+            count = (page + 1) * 10
+            if max_results != 0 and count > max_results:
+                page_size = 10 if max_results - (page) * 10 == 0 else max_results - (page) * 10
+            paginated_endpoint = f"{endpoint}&page_size={page_size}&page={page}"
 
-        if phantom.is_fail(ret_val):
-            return action_result.set_status(phantom.APP_ERROR, 'Error retrieving records: {0}'.format(action_result.get_message()))
+            ret_val, response, resp_data = self._make_rest_call(paginated_endpoint, action_result)
+
+            if phantom.is_fail(ret_val):
+                if PAGINATION_COMPLETE in action_result.get_message():
+                    break
```

Comment:
When does this happen vis a vis the case below where the action did not fail but pagination was completed?

Reply:
If we don't break, it may go for end of iteration in case where max_artifacts entered by used is greater than actual number of artifacts.